backend/src/nodes/utils/utils.py

```diff
@@ -187,88 +187,126 @@ def nptensor2np(
     return img_np.astype(imtype)
 
 
-def convenient_upscale(
-    img: np.ndarray,
-    input_channels: int,
-    upscale: Callable[[np.ndarray], np.ndarray],
-) -> np.ndarray:
-    """
-    Upscales the given image in an intuitive/convenient way.
-
-    This method guarantees that the `upscale` function will be called with an image with
-    `input_channels` number of channels.
-    """
-
+def to_target_channels(
+    img: np.ndarray, model_in_nc: int
+) -> Tuple[np.ndarray, np.ndarray | int | None]:
+    """Adjust number of image channels if it does not equal number of channels
+    required by the model."""
     c = get_h_w_c(img)[2]
 
-    # Transparency hack (white/black background difference alpha)
-    if c == 4 and input_channels in (1, 3):
-        # Ignore single-color alpha
+    imgout = img
+    supplemental_output = None
```

Comment:
You might want to call this "alpha" everywhere, cause that's what it is.

Reply:
It isn't alpha everywhere though. When using difference alpha, it's returning a modified copy of the original image which is then upscaled and used to calculate alpha later in the code.

Reply:
Right, we do this white/black background trick. Good point.